go/runtime/txpool/txpool.go

```diff
@@ -78,6 +78,17 @@ type TransactionPool interface {
 	// SubmitTxNoWait adds the transaction into the transaction pool and returns immediately.
 	SubmitTxNoWait(ctx context.Context, tx []byte, meta *TransactionMeta) error
 
+	// SubmitProposedBatch adds the given (possibly new) transaction batch into the current
+	// proposal queue.
+	SubmitProposedBatch(batch [][]byte)
+
+	// PromoteProposedBatch promotes the specified transactions that are already in the transaction
+	// pool into the current proposal queue.
+	PromoteProposedBatch(batch []hash.Hash)
+
+	// ClearProposedBatch clears the proposal queue.
+	ClearProposedBatch()
```

Comment:
This is the only thing that clears the proposal queue.  `TransactionPool.Clear` does not.  Is this intentional?

Reply:
`Clear` is not currently used anywhere, but yeah it makes sense that it should also clear the proposal queue. Will fix.